metal/datasource_metal_facility.go

```diff
@@ -50,8 +57,16 @@ func dataSourceMetalFacilityRead(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error listing Facilities: %s", err)
 	}
 
+	dfRaw, dfOk := d.GetOk("desired_features")
+
 	for _, f := range facilities {
 		if f.Code == code {
+			if dfOk {
+				unsupported := difference(convertStringArr(dfRaw.(*schema.Set).List()), f.Features)
+				if len(unsupported) > 0 {
+					return fmt.Errorf("facililty %s doesn't have feature(s) %v", f.Code, unsupported)
```

Comment:
Shouldn't we `continue` until we find a match?  Only error if there were no matches?

The logic has to be changed since we are no longer accepting the first match.

Reply:
The code is `Required`. If the code matches, then this is the fac we're looking for.

This PR doesn't return a list (or first found), it returns just one facility. It doesn't really filter, just checks that the found fac has the properties.

Reply:
I see. Having a code in mind reduces the usability of this new feature to me (I'm thinking about this as https://github.com/equinix/terraform-provider-metal/issues/180#issuecomment-909863036). 

This feature will still be helpful when validating that some user-supplied `var.facility` used in a module or configuration is a facility with the correct features.